### ISSUE TITLE
Aliases usage/sizing guidance & scaling limits

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#104456]
As we adapt Elasticsearch to larger and larger clusters (see e.g. #77466) I've noticed something of an upwards trend in issues related to high alias counts. I've particularly noticed cases where the user has many (thousands or more) aliases pointing to a single index, whereas the implementation in Elasticsearch is more targetted towards situations where there are relatively few aliases each of which points to a large number of indices.

I don't think we have any docs that give guidance about alias usage or sizing, nor do we have any safety limits on alias counts, and this makes it hard for users to design their system to use aliases in Elasticsearch appropriately.

**Comment by alexander-marquardt** [comment]
Is this issue fixed by https://github.com/elastic/elasticsearch/pull/115934 ? I have a user that reports that after upgrading, they do not experience rollover of a very active index is creating drops in ingestion rate cluster wide

**Comment by DaveCTurner** [comment]
No, we still lack docs giving guidance about the scaling limits around aliases. It may be that the situation is better than it was, but these things are still bounded.

Also please don't mention stuff from private repositories in the public Github repo.
